### ISSUE TITLE
RawPostDataException - cannot access request body

### ISSUE THREAD

**Issue by kennethklee** [Mashape/analytics-agent-python#2]
Please see note in https://docs.djangoproject.com/en/dev/topics/http/middleware/#process-view

This particular user is using `django-rest-framework`.

**The error**
RawPostDataException: You cannot access body after reading from request's data stream due to accessing request.body in your agent. 

**The location**

```
/home/api/.virtualenvs/api/local/lib/python2.7/site-packages/mashapeanalytics/middleware/django_middleware.py in process_response 
def process_response(self, request, response): 
startedDateTime = request.META.get('MASHAPE_ANALYTICS.STARTED_DATETIME', datetime.utcnow()) 
requestHeaders = [{'name': re.sub('^HTTP_', '', header), 'value': value} for (header, value) in request.META.items() if header.startswith('HTTP_')] 
requestHeaderSize = self.request_header_size(request) 
requestQueryString = [{'name': name, 'value': (value[0] if len(value) > 0 else None)} for name, value in parse_qs(request.META.get('QUERY_STRING', '')).items()] 
requestContentSize = len(request.body) ... 
responseHeaders = [{'name': header, 'value': value[-1]} for (header, value) in response._headers.items()] 
responseHeadersSize = self.response_header_size(response) 
responseContentSize = len(response.content) 
alf = Alf(self.serviceToken, self.environment, self.client_address(request))
```

**Comment by kennethklee** [comment]
This was fixed in f18a03b158726f386e82cb51f10fb334ef4a4a0f

**Comment by michaelvilensky** [comment]
I'm having the same problem in a middleware I'm writing.
Could it be the no middleware can access the body of a request?
I was trying to use all the options, body, read, readlines, non of them worked.
